### DISCUSSION TITLE
Partial list of agents scores

### DISCUSSION THREAD

**Discussion by eranhirs** [yasserfarouk/scml#94] [one-shot-qa] [answered]
After adding a few agents to the `competitors` list in `myagent.py` (from the skeleton), in the final table of agents I sometimes get a partial list.
Is it intentionally not showing all agents? I also sometimes get agents I don't have in my list, such as `SingleAgreementAspirationAgent`, where are they coming from?

**Comment by yasserfarouk** [comment, marked as answer]
This is unintended behavior. It should show all agents you give it and nothing else. We will consider this a bug. Can you open an issue with some minimal example to reproduce the issue? If you do not want to share code here, please just email me an example (yasserfarouk@gmail.com) and we will look into it.

`SingleAgreementAspirationAgent` is a builtin agent that we use by default to fill any factories that are not assigned to any other agents in the world simulation. This may be why you see it sometimes.

**Reply by yasserfarouk** [reply]
@eranhirs Please check issue #94 
It should be resolved in SCML 0.5.5
